executor/instruction/mod.rs

```diff
@@ -358,29 +365,96 @@ impl<T: Hkt> Checker<T> {
             (if select_a_min { a_min } else { b_min }, if select_a_max { a_max } else { b_max })
         }
 
-        /*
         let mut range = (Bound::Unbounded, Bound::Unbounded);
-        for check in &self.checks {
-            match *check {
-                CheckInstruction::Comparison { lhs, rhs, comparator } if lhs == Vertex::Variable(target) => {
-                    let rhs = row.get(rhs).to_owned();
-                    let comp_range = match comparator {
-                        Comparator::Equal => (Bound::Included(rhs.clone()), Bound::Included(rhs)),
-                        Comparator::Less => (Bound::Unbounded, Bound::Excluded(rhs)),
-                        Comparator::LessOrEqual => (Bound::Unbounded, Bound::Included(rhs)),
-                        Comparator::Greater => (Bound::Excluded(rhs), Bound::Unbounded),
-                        Comparator::GreaterOrEqual => (Bound::Included(rhs), Bound::Unbounded),
-                        Comparator::Like => continue,
-                        Comparator::Contains => continue,
-                    };
-                    range = intersect(range, comp_range);
+        for i in 0..self.checks.len() {
+            let check = &self.checks[i];
+            match check {
+                CheckInstruction::Comparison { lhs, rhs, comparator } => {
+                    if lhs.as_variable().is_some_and(|var| var == target_variable) {
```

Comment:
This is a little cleaner imo:
```suggestion
                    if lhs.as_variable() == Some(target_variable) {
```

Reply:
Done